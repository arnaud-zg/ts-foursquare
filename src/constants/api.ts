export const CLIENT_ID_KEY = 'apiClientId'
export const CLIENT_SECRET_KEY = 'apiClientSecret'

export interface ICredentials {
  [CLIENT_ID_KEY]: string
  [CLIENT_SECRET_KEY]: string
}

export enum EApiDefaultParameters {
  VERSION = '20190425',
}
