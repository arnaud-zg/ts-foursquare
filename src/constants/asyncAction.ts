import { getType } from 'typesafe-actions'
import {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../DEPRECATED_actions'

interface IAsyncActionNameMapping {
  [actionType: string]: string
}

export const ASYNC_ACTION_NAME_MAPPING: IAsyncActionNameMapping = {
  // getVenuesCategoriesAsync
  [getType(getVenuesCategoriesAsync.request)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.success)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.failure)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.cancel)]: 'getVenuesCategoriesAsync',
  // getVenuesExploreAsync
  [getType(getVenuesExploreAsync.request)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.success)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.failure)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.cancel)]: 'getVenuesExploreAsync',
  // getVenuesLikesAsync
  [getType(getVenuesLikesAsync.request)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.success)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.failure)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.cancel)]: 'getVenuesLikesAsync',
  // getVenuesListedAsync
  [getType(getVenuesListedAsync.request)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.success)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.failure)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.cancel)]: 'getVenuesListedAsync',
  // getVenuesNextVenuesAsync
  [getType(getVenuesNextVenuesAsync.request)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.success)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.failure)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.cancel)]: 'getVenuesNextVenuesAsync',
  // getVenuesSearchAsync
  [getType(getVenuesSearchAsync.request)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.success)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.failure)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.cancel)]: 'getVenuesSearchAsync',
  // getVenuesSimilarAsync
  [getType(getVenuesSimilarAsync.request)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.success)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.failure)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.cancel)]: 'getVenuesSimilarAsync',
  // getVenuesSuggestCompletionAsync
  [getType(
    getVenuesSuggestCompletionAsync.request
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.success
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.failure
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.cancel
  )]: 'getVenuesSuggestCompletionAsync',
  // getVenuesTrendingAsync
  [getType(getVenuesTrendingAsync.request)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.success)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.failure)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.cancel)]: 'getVenuesTrendingAsync',
}
