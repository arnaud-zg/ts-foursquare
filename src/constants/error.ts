export enum ECommonError {
  RESPONSE_NOT_VALID = 'Response is not valid',
}
