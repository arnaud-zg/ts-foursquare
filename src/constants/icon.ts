export enum EIconSize {
  S = '32',
  M = '44',
  L = '64',
  XL = '88',
}
