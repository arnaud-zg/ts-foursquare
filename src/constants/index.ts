export const PROJECT_NAME = 'ts-foursquare'
