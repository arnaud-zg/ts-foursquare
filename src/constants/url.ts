export enum EUrlSeparator {
  LOCATION_SEARCH_SEPARATOR = '?',
  QUERY_STRING_SEPARATOR = '&',
}
