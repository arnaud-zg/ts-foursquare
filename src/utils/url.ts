interface ILocationSearchParam {
  [key: string]: string | number | boolean | undefined
}

export const getLocationSearch = (
  { param = {} }: { param: ILocationSearchParam } = { param: {} }
) => {
  const locationSearch = Object.keys(param)
    .filter(key => param[key] !== undefined && param[key] !== '')
    .map(
      key => `${encodeURIComponent(key)}=${encodeURIComponent(`${param[key]}`)}`
    )
    .join('&')

  return (!!locationSearch && `?${locationSearch}`) || ''
}

export const getLocationHref = ({
  origin,
  pathname,
  param = {},
}: {
  origin: string
  pathname: string
  param?: ILocationSearchParam
}) => `${origin}${pathname}${getLocationSearch({ param })}`
