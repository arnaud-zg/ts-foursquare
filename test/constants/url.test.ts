import { EUrlSeparator } from '../../src/constants/url'

describe('constants/url/EUrlSeparator', () => {
  it('should take a snapshot of EUrlSeparator', () => {
    expect(EUrlSeparator).toMatchSnapshot()
  })
})
