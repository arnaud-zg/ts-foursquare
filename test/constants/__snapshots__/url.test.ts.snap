// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants/url/EUrlSeparator should take a snapshot of EUrlSeparator 1`] = `
Object {
  "LOCATION_SEARCH_SEPARATOR": "?",
  "QUERY_STRING_SEPARATOR": "&",
}
`;
